### DISCUSSION TITLE
Compiling Lilypond on Termux

### DISCUSSION THREAD

**Discussion by nagymusic** [termux/termux-app#4238] [Q&A]
I successfully installed Lilypond music notation software using Termux on my Android (version 12) tablet, but I can't view the PDF after compiling the file.

Here's the error message I received (I highlighted the error message I'm trying to understand):

```
~ $ lilypond [[myfile.ly](http://myfile.ly/)](http://myfile.ly/)
GNU LilyPond 2.24.4 (running Guile 3.0)
Processing `[[myfile.ly](http://myfile.ly/)](http://myfile.ly/)'
Parsing...
Interpreting music...
Preprocessing graphical objects...
Finding the ideal number of pages...
Fitting music on 1 page...
Drawing systems...
/data/data/com.termux/files/usr/share/lilypond/2.24.4/ly/init.ly:65:2: error: Guile signaled an error for the expression beginning here
#
 (let ((book-handler (if (defined? 'default-toplevel-book-handler)
In procedure open_iconv_descriptors: invalid or unknown character encoding LATIN1
```

Do you have any suggestions on how to fix this issue, or where can I seek help?

Thanks so much!

**Comment by JackMorningStar** [comment]
Hi,

I have already faced this issue from my termux and I got the solution from
my system when I raised this concern over the blackbox, and I got the
solution from there.

I hope you will find the same way.

Regards
Farmaan

**Reply by rwl7532** [reply]
Same error from Guile re: LATIN1. Gemini AI suggested exports (language settings) but it did not work. What was your solution?

**Comment by ddtopham** [comment]
I have same problem and have not found solution.  Is the Scheme code in init.ly incorrect? Or is there something missing that causes LATIN1 to not be found? I tried converting the ly file to utf8 using iconv and vi, but always get same error from Guile.

**Comment by ddtopham** [comment]
I saw online a suggestion to put this at top of my lilypond file:

%!unicorn

Now I don't get the error about latin!

But I don't see any pdf either...

**Reply by rwl7532** [reply]
Didn't work for me. %!unicorn looks like a lilypond comment. Long convo with Gemini ends with "let's see what github comments say"

**Comment by ddtopham** [comment]
Sorry, meant #!unicorn
True it is comment, which I believe guile uses, and the latin1 error was a
guile error

**Reply by rwl7532** [reply]
Still no happiness.  Simple .ly code
\version "2.24.4"

 \score
 {

{
  c'1
  }

 }

After some #!unicorn  variations  the worst was
Unbound variable: !
mytest.ly:6:2: error: syntax error, unexpected \score, expecting '.' or '='

 \score

**Comment by ddtopham** [comment]
I believe another form of comment may work...
Try

%!unicode

**Reply by rwl7532** [reply]
unicode and unicorn are a horse of a different color or horn.

%!unicode at the top line causes no mention of not understanding by Guile or lilypond. 
The outcome is still ending with LATIN1 
NO JOY

**Comment by ddtopham** [comment]
I get no error, but no file produced either

~ $ vi tryit.ly ~ $ lilypond tryit.ly GNU LilyPond 2.24.4 (running Guile
3.0) Processing `tryit.ly' Parsing... Success: compilation successfully
completed
